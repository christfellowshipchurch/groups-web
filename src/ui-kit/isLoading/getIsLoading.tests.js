import React from 'react';
import renderer from 'react-test-renderer';

// import Providers from '../../Providers';

import { getIsLoading, IsLoadingContext } from '.';

describe('the getIsLoading HOC', () => {
  it('should return false from context', () => {
    const Context = getIsLoading(({ isLoading }) => isLoading.toString());

    const tree = renderer.create(
      <IsLoadingContext.Provider value={false}>
        <Context />
      </IsLoadingContext.Provider>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should return true from context', () => {
    const Context = getIsLoading(({ isLoading }) => isLoading.toString());

    const tree = renderer.create(
      <IsLoadingContext.Provider value>
        <Context />
      </IsLoadingContext.Provider>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should return the isLoading value (true) from the component', () => {
    const Context = getIsLoading(({ isLoading }) => isLoading.toString());

    const tree = renderer.create(
      <IsLoadingContext.Provider>
        <Context isLoading />
      </IsLoadingContext.Provider>
    );

    expect(tree).toMatchSnapshot();
  });
});
