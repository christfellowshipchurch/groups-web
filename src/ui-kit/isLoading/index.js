import React, { useContext } from 'react';

// Create our IsLoadingContext
const IsLoadingContext = React.createContext(false);

/* Safely checks for isLoading to make it easy to build components that respond to either props or
 * context for rendering placeholder/loading state. */
const getIsLoading = (Component) => (props) => {
  // does the component have an isLoading property?
  if (Object.hasOwnProperty.call(props, 'isLoading')) {
    return <Component {...props} />;
  }

  // No? That's fine we'll pull it from context
  const isLoading = useContext(IsLoadingContext);

  return <Component isLoading={isLoading} {...props} />;
};

// Allows a component to accept an `isLoading` prop to pass down into context for it's children.
const withIsLoading = (Component) => ({ isLoading, ...props }) => (
  <IsLoadingContext.Provider value={isLoading}>
    <Component {...props} />
  </IsLoadingContext.Provider>
);

export { getIsLoading, IsLoadingContext, withIsLoading };
