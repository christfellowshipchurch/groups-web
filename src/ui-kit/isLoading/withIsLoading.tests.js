import React from 'react';
import renderer from 'react-test-renderer';

import { getIsLoading, withIsLoading } from '.';

describe('the withIsLoading HOC', () => {
  it('should render', () => {
    const ParentComponentPassingConext = withIsLoading(
      ({ children }) => children
    );

    const tree = renderer.create(
      <ParentComponentPassingConext>
        <h1>Boom</h1>
      </ParentComponentPassingConext>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should pass the isLoading prop (true) to context for children to consume', () => {
    const ChildComponentGettingContext = getIsLoading(({ isLoading }) =>
      isLoading.toString()
    );

    const ParentComponentPassingConext = withIsLoading(() => (
      <ChildComponentGettingContext />
    ));

    const tree = renderer.create(<ParentComponentPassingConext isLoading />);

    expect(tree).toMatchSnapshot();
  });
});
