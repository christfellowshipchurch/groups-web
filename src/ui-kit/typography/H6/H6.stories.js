import React from 'react';
import { storiesOf } from '@storybook/react';

import H6 from '.';

storiesOf('ui-kit/typography/H6', module)
  .add('default', () => (
    <H6>
      {
        '"God is most glorified in us when we are most satisfied in Him" ― John Piper'
      }
    </H6>
  ))
  .add('isLoading', () => (
    <H6 isLoading showLoadingAnimation>
      {
        '"God is most glorified in us when we are most satisfied in Him" ― John Piper'
      }
    </H6>
  ))
  .add('withMargins', () => <H6 withMargins>Padded Heading Big and Long</H6>)
  .add('withMargins + isLoading', () => (
    <H6 isLoading withMargins showLoadingAnimation>
      {
        '"God is most glorified in us when we are most satisfied in Him" ― John Piper'
      }
    </H6>
  ));
