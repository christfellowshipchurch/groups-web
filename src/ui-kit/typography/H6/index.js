import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import Placeholder from '../../Placeholder';

const h6Style = css`
  h6,
  .H6 {
    color: ${({ theme }) => theme.colors.text.secondary};
    font-family: ${({ theme }) => theme.typography.fontFamily};
    font-size: 0.75rem;
    font-weight: 500;
    line-height: ${({ theme }) => theme.helpers.verticalRhythm(0.75)};
    &.custom-placeholder {
      height: 0.75rem; /* Should match .h6font-size */
    }
  }
`;

const StyledH6 = styled.h6.attrs({
  className: 'ui-kit-typography-H6-StyledH6',
})`
  margin: ${({ padded, theme }) =>
    padded ? `0 0 ${theme.helpers.verticalRhythm(0.375)} 0` : 'initial'};
`;

const H6 = ({ children, isLoading, padded, ...props }) => (
  <Placeholder.Typography
    isLoading={isLoading}
    showLoadingAnimation
    typographicClassName={'H6'} // needs to match child components className
  >
    <StyledH6 padded={padded} {...props}>
      {children}
    </StyledH6>
  </Placeholder.Typography>
);

H6.defaultProps = {
  isLoading: false,
};

H6.propTypes = {
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool,
  padded: PropTypes.bool,
};

H6.displayName = 'ui-kit.typography.H6';

export { H6 as default, h6Style };
