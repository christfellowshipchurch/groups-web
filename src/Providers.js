import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
