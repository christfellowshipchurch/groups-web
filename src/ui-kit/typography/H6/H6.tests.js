import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../../../Providers';
import { IsLoadingContext } from '../../isLoading';

import H6 from '.';

describe('the H6 component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <H6>Default H6 text</H6>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const salmonText = { color: 'salmon' };
    const tree = renderer.create(
      <Providers>
        <H6 style={salmonText}>Salmon text</H6>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with vertical rhythm (padded)', () => {
    const tree = renderer.create(
      <Providers>
        <H6 padded>H6 text with vertical rhythm styled via margin.</H6>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state', () => {
    const tree = renderer.create(
      <Providers>
        <H6 isLoading>Default H6 text</H6>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state via context', () => {
    const tree = renderer.create(
      <Providers>
        <IsLoadingContext.Provider value>
          <H6>Default H6 text</H6>
        </IsLoadingContext.Provider>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state with vertical rhythm (padded)', () => {
    const tree = renderer.create(
      <Providers>
        <H6 isLoading padded>
          Default H6 text
        </H6>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
