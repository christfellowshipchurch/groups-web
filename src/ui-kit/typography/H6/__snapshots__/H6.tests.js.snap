// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the H6 component should accept and render passed in styles 1`] = `
.c0 {
  color: #505050;
  font-size: 0.75rem;
  font-weight: 500;
  line-height: 1.125rem;
  margin: initial;
}

<h6
  className="c0 toneu3-2 ctXJce"
>
  Salmon text
</h6>
`;

exports[`the H6 component should render 1`] = `
.c0 {
  color: #505050;
  font-size: 0.75rem;
  font-weight: 500;
  line-height: 1.125rem;
  margin: initial;
}

<h6
  className="c0 toneu3-2 ctXJce"
>
  Default H6 text
</h6>
`;

exports[`the H6 component should render a loading state 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
}

.c1 {
  height: 0.75rem;
  margin: 0 0 calc(1.125rem - 0.75rem) 0;
}

<mark
  className="c0 show-loading-animation ra2rjz-1 c1 custom-placeholder"
/>
`;

exports[`the H6 component should render a loading state via context 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
}

.c1 {
  height: 0.75rem;
  margin: 0 0 calc(1.125rem - 0.75rem) 0;
}

<mark
  className="c0 show-loading-animation ra2rjz-1 c1 custom-placeholder"
/>
`;

exports[`the H6 component should render a loading state with vertical rhythm (withMargins) 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
}

.c1 {
  height: 0.75rem;
  margin: 0 0 0.5625rem 0;
}

<mark
  className="c0 show-loading-animation ra2rjz-1 c1 custom-placeholder"
/>
`;

exports[`the H6 component should render with vertical rhythm (withMargins) 1`] = `
.c0 {
  color: #505050;
  font-size: 0.75rem;
  font-weight: 500;
  line-height: 1.125rem;
  margin: 0 0 0.5625rem 0;
}

<h6
  className="c0 toneu3-2 ctXJce"
>
  H6 text with vertical rhythm styled via margin.
</h6>
`;
